File: Problems/0027_Remove_Element/Remove_Element.cs
using System;

public class Solution {
	public int RemoveElement(int[] nums, int val)
	{
		int i = 0;
		int n = nums.Length;
		while (i < n) {
			if (nums[i] == val) {
				nums[i] = nums[n - 1];
				// reduce array size by one
				n--;
			} else {
				i++;
			}
		}
		return n;
	}

	public void Main()
	{
		Console.Write(RemoveElement(new int[] { 3, 2, 2, 3 }, 2 ));
	}
}

File: Problems/0001_0099/0027_Remove_Element/Project_Python3/Remove_Element.py
import os
import sys
import time
from typing import List, Dict, Tuple

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        # 34ms - 37ms
        while val in nums:
            nums.remove(val)
        return len(nums)

def main():
    argv = sys.argv
    argc = len(argv)

    if argc < 2:
        print("Usage: python {0} <testdata.txt>".format(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("{0} not found...".format(argv[1]))
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        temp = temp.strip()
        if temp == "":
            continue
        print("args = {0}".format(temp))
        loop_main(temp)
    #   print("Hit Return to continue...")
    #   input()

def loop_main(temp):
    str_args = temp.replace("\"","").replace("[[","").replace("]]","").rstrip()
    flds = str_args.split("],[")

    nums = [int(n) for n in flds[0].split(",")]
    val = int(flds[1])
    print("nums = {0}, val = {1:d}".format(nums, val))

    sl = Solution()
    time0 = time.time()

    result = sl.removeElement(nums, val)

    time1 = time.time()

    print("result = {0}".format(result))
    print("Execute time ... : {0:f}[s]\n".format(time1 - time0))

if __name__ == "__main__":
    main()
